### ISSUE TITLE
Split query exception

### ISSUE THREAD

**Issue by chaithu322** [DuendeSoftware/Support#399]
**Which version of Duende BFF are you using?**
6

**Which version of .NET are you using?**
6.0.4

**Describe the bug**
I am receiving following exceptions in the log
An exception occurred 
while iterating over the results of a query for context type
'"Duende.IdentityServer.EntityFramework.DbContexts.ConfigurationDbContext"'." 
  ""System.Threading.Tasks.TaskCanceledException: A task was canceled.   
    at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.OpenInternalAsync(Boolean errorsExpected, CancellationToken cancellationToken)    
        at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.OpenInternalAsync(Boolean errorsExpected, CancellationToken cancellationToken)  
           at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.OpenAsync(CancellationToken cancellationToken, Boolean errorsExpected)   
                at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken) 
                    at Microsoft.EntityFrameworkCore.Query.Internal.SplitQueryingEnumerable`1.AsyncEnumerator.InitializeReaderAsync(AsyncEnumerator enumerator, CancellationToken cancellationToken)
                            at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)   
                                at Microsoft.EntityFrameworkCore.Query.Internal.SplitQueryingEnumerable`1.AsyncEnumerator.MoveNextAsync()"

There is no user experience error but below error it is happening
An exception occurred 
while iterating over the results of a query for context type
'"Duende.IdentityServer.EntityFramework.DbContexts.ConfigurationDbContext"'." 
  ""System.Threading.Tasks.TaskCanceledException: A task was canceled.   
    at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.OpenInternalAsync(Boolean errorsExpected, CancellationToken cancellationToken)    
        at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.OpenInternalAsync(Boolean errorsExpected, CancellationToken cancellationToken)  
           at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.OpenAsync(CancellationToken cancellationToken, Boolean errorsExpected)   
                at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken) 
                    at Microsoft.EntityFrameworkCore.Query.Internal.SplitQueryingEnumerable`1.AsyncEnumerator.InitializeReaderAsync(AsyncEnumerator enumerator, CancellationToken cancellationToken)
                            at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)   
                                at Microsoft.EntityFrameworkCore.Query.Internal.SplitQueryingEnumerable`1.AsyncEnumerator.MoveNextAsync()"

**To Reproduce**

Steps to reproduce the behavior.

**Expected behavior**
The logs should not have any exceptions

**Log output/exception with stacktrace**
An exception occurred 
while iterating over the results of a query for context type
'"Duende.IdentityServer.EntityFramework.DbContexts.ConfigurationDbContext"'." 
  ""System.Threading.Tasks.TaskCanceledException: A task was canceled.   
    at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.OpenInternalAsync(Boolean errorsExpected, CancellationToken cancellationToken)    
        at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.OpenInternalAsync(Boolean errorsExpected, CancellationToken cancellationToken)  
           at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.OpenAsync(CancellationToken cancellationToken, Boolean errorsExpected)   
                at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken) 
                    at Microsoft.EntityFrameworkCore.Query.Internal.SplitQueryingEnumerable`1.AsyncEnumerator.InitializeReaderAsync(AsyncEnumerator enumerator, CancellationToken cancellationToken)
                            at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)   
                                at Microsoft.EntityFrameworkCore.Query.Internal.SplitQueryingEnumerable`1.AsyncEnumerator.MoveNextAsync()"

```
data
```

**Additional context**

No other context i am aware of

**Comment by josephdecock** [comment]
These cancelation exceptions are really just noise and you don't need to worry about them. What's happening is, IdentityServer passes cancellation tokens into the entity framework calls that it makes. Those cancellation tokens come from ASP.NET's pipeline, so if an http connection is terminated before the response is complete, that cancellation gets propagated into EF. The good news about that is, EF can cancel its request to the database and reduce load on the db. The bad news is that entity framework then throws and logs this exception. What we've seen is that both the SQL Server EF provider and EF itself will log that exception as an error. Then IdentityServer also catches unhandled exceptions and logs those, so you sometimes get triple error messages in the logs, even though http connections being terminated is actually an expected occurrence. What we've done to mitigate this is, we've added a configuration option in version 6.2 that filters out exceptions from IdentityServer's logging. The default is to filter these TaskCanceledExceptions. We also got in touch with the EF folks, and they agree that they shouldn't be logging at this level. In the newest release of EF, they've reduced the severity of the logs, but only in one of the two places.
